cmake/get_compiler_info.cmake

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) Open Enclave SDK contributors.
+# Licensed under the MIT License.
+#
+# This is a helper function to obtain the compiler information
+# (e.g., the GNU or clang compiler and its version) that is used
+# to generate the pre-built enclave test cases.
+#
+function(get_compiler_info)
+    # Check if the path to the pre-built test cases is specified.
+    if (LINUX_BIN_DIR)
+        # Search the cmake file that includes the compiler infomation
+        # in the auto-generated file from pre-built cmake directories.
+        set(target_path ${LINUX_BIN_DIR}/../CMakeFiles)
```

Comment:
This is an unsafe directory traversal; it's based on an assumption of what the CI/CD pipeline copies and sets the `LINUX_BIN_DIR` today that can easily be broken. 

Consider an explicit contract between the test enclave build process and the test host build process instead:
- tests/libcxx/enc/CMakeLists.txt generates a record of its compiler configuration to disk
- tests/libcxx/CMakeLists.txt probes for the record given its relative path under the `LINUX_BIN_DIR` and loads the values if found.

Reply:
Updated based on your suggestion.